optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -421,7 +421,8 @@ public ScoreDirectorFactoryConfig withAssertionScoreDirectorFactory(
             KieResources kieResources = kieServices.getResources();
             KieFileSystem kieFileSystem = kieServices.newKieFileSystem();
             if (!ConfigUtils.isEmptyCollection(scoreDrlList)) {
-                ClassLoader actualClassLoader = (classLoader != null) ? classLoader : getClass().getClassLoader();
+                ClassLoader actualClassLoader =
+                        (classLoader != null) ? classLoader : Thread.currentThread().getContextClassLoader();
```

Comment:
We do indeed need to use the Thread.currentThread().getContextClassLoader() in quarkus (Stuart Douglas did a simiar change). Great that you've find the fix.

I'd rather not change the default ScoreDirectorFactoryConfig logic for that - it can have side effects in other enviromnents, but instead we need to make sure that quarkus extension passes it's classloader.

Reply:
@Christopher-Chianelli Can you try making this change in OptaPlannerBeanProvider instead?
All those create() methods called there, also accept a ClassLoader